vault/core.go

```diff
@@ -2939,3 +2944,34 @@ type LicenseState struct {
 	ExpiryTime time.Time
 	Terminated bool
 }
+
+type InFlightReqData struct {
+	StartTime        time.Time `json:"start_time"`
+	ClientRemoteAddr string    `json:"client_remote_address"`
+	ReqPath          string    `json:"request_path"`
+	Duration         string    `json:"duration"`
+}
+
+func (c *Core) StoreInFlightReqData(reqID string, inFlightReqdata *InFlightReqData) error {
+	if c.inFlightReqMap == nil {
```

Comment:
It shouldn't be possible for this to ever happen, right?  A core can't be created with this nil?  So let's not test for it.